lib/services/server.go

```diff
@@ -985,3 +991,9 @@ func GuessProxyHostAndVersion(proxies []Server) (string, string, error) {
 	guessProxyHost := fmt.Sprintf("%v:%v", proxies[0].GetHostname(), defaults.HTTPListenPort)
 	return guessProxyHost, proxies[0].GetTeleportVersion(), nil
 }
+
+// validKubeClusterName filters the allowed characters in kubernetes cluster
+// names. We need this because cluster names are used for cert filenames on the
+// client side, in the ~/.tsh directory. Restricting characters helps with
+// sneaky cluster names being used for client directory traversal and exploits.
+var validKubeClusterName = regexp.MustCompile(`^[a-zA-Z0-9._-]+$`)
```

Comment:
shouldn't this align with the [k8s](https://github.com/kubernetes/apimachinery/blob/release-1.17/pkg/api/validation/objectmeta.go#L43) validator?

Reply:
I don't think we need to force their format. This cluster name shouldn't be directly used anywhere on the k8s API side.
Our regexp is a bit more lenient, so users can still use their k8s-approved cluster names.